### PR TITLE
Upgrade to v2.87.0 of the AzureRM Terraform Provider

### PR DESCRIPTION
**Pull request by stack72**
Fixes: #986﻿

### CODE REVIEW

File: sdk/go/azure/compute/managedDisk.go [hunk 1]
```diff
@@ -145,6 +145,8 @@ type ManagedDisk struct {
 	OnDemandBurstingEnabled pulumi.BoolPtrOutput `pulumi:"onDemandBurstingEnabled"`
 	// Specify a value when the source of an `Import` or `Copy` operation targets a source that contains an operating system. Valid values are `Linux` or `Windows`.
 	OsType pulumi.StringPtrOutput `pulumi:"osType"`
+	// Whether it is allowed to access the disk via public network. Defaults to `true`.
+	PublicNetworkAccessEnabled pulumi.BoolPtrOutput `pulumi:"publicNetworkAccessEnabled"`
```
Comment (guineveresaenger) [inline sdk/go/azure/compute/managedDisk.go:149]: I'm assuming this is something new from the pulumi side?

File: sdk/go/azure/compute/orchestratedVirtualMachineScaleSet.go [hunk 2]
```diff
@@ -71,7 +69,10 @@ type OrchestratedVirtualMachineScaleSet struct {
 	DataDisks               OrchestratedVirtualMachineScaleSetDataDiskArrayOutput           `pulumi:"dataDisks"`
 	EncryptionAtHostEnabled pulumi.BoolPtrOutput                                            `pulumi:"encryptionAtHostEnabled"`
 	EvictionPolicy          pulumi.StringPtrOutput                                          `pulumi:"evictionPolicy"`
-	Identity                OrchestratedVirtualMachineScaleSetIdentityPtrOutput             `pulumi:"identity"`
+	Extensions              OrchestratedVirtualMachineScaleSetExtensionArrayOutput          `pulumi:"extensions"`
+	// Specifies the time alloted for all extensions to start. The time duration should be between 15 minutes and 120 minutes (inclusive) and should be specified in ISO 8601 format. The default value is 90 minutes (PT1H30M).
```
Comment (guineveresaenger) [inline sdk/go/azure/compute/orchestratedVirtualMachineScaleSet.go:73]: s/alloted/allotted

File: sdk/go/azure/compute/orchestratedVirtualMachineScaleSet.go [hunk 3]
```diff
@@ -71,7 +69,10 @@ type OrchestratedVirtualMachineScaleSet struct {
 	DataDisks               OrchestratedVirtualMachineScaleSetDataDiskArrayOutput           `pulumi:"dataDisks"`
 	EncryptionAtHostEnabled pulumi.BoolPtrOutput                                            `pulumi:"encryptionAtHostEnabled"`
 	EvictionPolicy          pulumi.StringPtrOutput                                          `pulumi:"evictionPolicy"`
-	Identity                OrchestratedVirtualMachineScaleSetIdentityPtrOutput             `pulumi:"identity"`
+	Extensions              OrchestratedVirtualMachineScaleSetExtensionArrayOutput          `pulumi:"extensions"`
```
Comment (guineveresaenger) [inline sdk/go/azure/compute/orchestratedVirtualMachineScaleSet.go:72]: Same thing - related to a Pulumi addition?

File: sdk/go/azure/compute/managedDisk.go (hunk 1 above)
Comment (stack72) [reply sdk/go/azure/compute/managedDisk.go:149]: This is an addition upstream

File: sdk/go/azure/compute/orchestratedVirtualMachineScaleSet.go (hunk 2 above)
Comment (stack72) [reply sdk/go/azure/compute/orchestratedVirtualMachineScaleSet.go:73]: This is an upstream comment we have no real control over

File: sdk/go/azure/compute/orchestratedVirtualMachineScaleSet.go (hunk 3 above)
Comment (stack72) [reply sdk/go/azure/compute/orchestratedVirtualMachineScaleSet.go:72]: Yup!

File: sdk/go/azure/sentinel/authomationRule.go [hunk 4]
```diff
@@ -55,12 +55,12 @@ import (
 // 		if err != nil {
 // 			return err
 // 		}
-// 		_, err = sentinel.NewAuthomationRule(ctx, "exampleAuthomationRule", &sentinel.AuthomationRuleArgs{
+// 		_, err = sentinel.NewAutomationRule(ctx, "exampleAutomationRule", &sentinel.AutomationRuleArgs{
```
Comment (guineveresaenger) [inline sdk/go/azure/sentinel/authomationRule.go:58]: Do we still need this file, given the correctly spelled one is generated below?

File: sdk/go/azure/iot/endpointEventhub.go [hunk 5]
```diff
@@ -68,7 +68,6 @@ import (
 // 			Location:          exampleResourceGroup.Location,
 // 			Sku: &iot.IoTHubSkuArgs{
 // 				Name:     pulumi.String("B1"),
-// 				Tier:     "Basic",
```
Comment (guineveresaenger) [inline sdk/go/azure/iot/endpointEventhub.go:71]: Is this a breaking change?

File: sdk/go/azure/sentinel/pulumiTypes.go [hunk 6]
```diff
@@ -899,6 +899,387 @@ func (o AuthomationRuleConditionArrayOutput) Index(i pulumi.IntInput) Authomatio
 	}).(AuthomationRuleConditionOutput)
 }
 
+type AutomationRuleActionIncident struct {
```
Comment (guineveresaenger) [inline sdk/go/azure/sentinel/pulumiTypes.go:902]: I assume the typo migration will happen in the next release?

File: provider/resource.go [hunk 7]
```diff
@@ -2592,6 +2598,11 @@ func Provider() tfbridge.ProviderInfo {
 		azureResource(azureContainerService, "RegistryWebook"),
 		azureResource(azureContainerService, "RegistryWebhook"), azureContainerService, azureContainerService, nil)
 
+	// fix spelling of sentinel.AutomationRule
```
Comment (guineveresaenger) [inline provider/resource.go:2601]: I assume the rename works as advertised - would the misspelled rerource be removed in the next release? or are we providing for backwards compatibility here?

**Review by guineveresaenger — COMMENTED** [review]
Seems legit to me - 
I wonder about the `pulumi.StrPtrInput`s that are sprinkled throughout - since they're a string field, I hope they add extra capability rather than breaking anything.

**Pull request merged**